samples/nrf9160/gnss/src/main.c

```diff
@@ -163,6 +163,19 @@ static void agps_data_get(struct k_work *item)
 
 	int err;
 
+	/* With minimal assistance, assistance is injected only when GNSS starts for the
```

Comment:
Assistance is only used for the first fix, after that GNSS knows where it is. If assistance would be injected in one location and after that the device would move thousands of kilometers before GNSS is started, then it would be searching for the wrong satellites. But that's not how the sample works.

The MCC based locations are anyway very coarse. The maximum uncertainty radius for the location is 1800 km.

And if you meant about roaming, the MCC identifies the country of the network the device is registered in. So, it works fine also when roaming.

Reply:
Yeah, of course the GNSS keeps track and updates constantly the last known position as it gets fixes, silly me. I was somehow thinking that the once injected position is used over and over again.

Reply:
> But that's not how the sample works.

If an application has functionality to turn GNSS tracking off and on again, would you write this app differently from minimal assistance perspective? I'm trying to find out if customers will be directed to inaccurate implementation in some cases and they might not realize it.

This comment about when minimal assistance is injected into GNSS could be mentioned in the document and maybe some limitations from this discussion. Or did I miss it from there?
Do we have some other document than this sample which talks about which assistance methods should be used in different use cases?

Reply:
As with all types of assistance, the assistance data should be injected when it's going to be used. For example, ephemerides will go stale in a few hours, and would be of no use anymore after that. The sample implementation is such, that all assistance data is injected when GNSS asks for it. So, GNSS is first started and then sends the `NRF_MODEM_GNSS_EVT_AGPS_REQ` event to indicate that it needs assistance data, after which the assistance data is possibly downloaded and injected.

This is the normal way of using assistance. GNSS sends the `NRF_MODEM_GNSS_EVT_AGPS_REQ` event whenever it would benefit from fresh assistance data and the application acts on it.

We don't really have documentation discussing the different assistance methods and their pros and cons yet. That's something that we should have in the future, but in my opinion it's out of scope of this PR and also the GNSS sample. There's no simple answer, but basically, A-GPS is the preferred method unless for some reason it can't be used. And almanac + time + location should be better (at least in some cases) than no assistance at all, but we don't yet have any numbers for that. P-GPS is intended more for some special use cases and it might make sense to also combine it with A-GPS.

Reply:
I created NCSDK-12900 to add better documentation about GNSS assistance methods.